### DISCUSSION TITLE
How to open the extension page on iOS? (So I can import a remote script)

### DISCUSSION THREAD

**Discussion by quinncomendant** [quoid/userscripts#654] [Q&A] [answered]
The Userscripts extension on macOS has a link to “Open Extension Page”:

![SCR-20240512-ptky](https://github.com/quoid/userscripts/assets/6884276/53b2d27e-9335-4dc4-be43-5b274ea849b8)

However, when activating the extension on iOS, that link is absent:

![IMG_6868](https://github.com/quoid/userscripts/assets/6884276/a0647737-16a4-4a07-ac2b-2852dd7d1845)

**How do I open the script editor on iOS?**

At first, I wanted to access the extension page to import a remote script. Since I was unable to do that, I was able to add the script on macOS and sync it to iOS using iCloud. However, now I would like to edit that script on iOS, but I can't find a way to open the script in the editor. I understand this should be possible? (The README.md doesn't say it is not possible.)

Another point of confusion: The README.md mentions “[…] open the containing app”. What is a containing app? I'm a software developer myself, and I'm not familiar with that nomenclature. Does it mean the Userscripts app itself (which exports the extension to the OS)?

When I open the Userscripts app, this is what I see:

![IMG_6869](https://github.com/quoid/userscripts/assets/6884276/e2757005-feba-432d-899c-3166b7298188)

**Comment by ACTCD** [comment, marked as answer]
> I understand this should be possible? (The README.md doesn't say it is not possible.)

The iOS version does not include the script editor provided in the macOS version.

But we did add a settings page similar to the macOS version in the latest [beta version](https://github.com/quoid/userscripts/releases).

> The README.md mentions “[…] open the containing app”. What is a containing app?...

Yes, you understood correctly. It's called from [Apple](https://developer.apple.com/library/archive/documentation/General/Conceptual/ExtensibilityPG/ExtensionOverview.html).

> So I can import a remote script

Visit any `.user.js` URL in Safari, open the extension `popup` and you will see an installation prompt.

You can also save scripts directly to the Userscripts directory you set.

Both of the above work equally well in the macOS version.

> How do I open the script editor on iOS?

Anyway, you can edit script files in the folder you specify directly on iOS. (use any third-party code editor Apps)

**Comment by quinncomendant** [comment]
Great explanation @ACTCD, thank you. Can we convert this discussion to an issue to update the documentation so others aren't as confused as I was? The specific points that need to be added are:

- iOS version does not include a script editor, only macOS version does.
- How to import a remote script to the iOS version:
  - Visit any `.user.js` URL in Safari, open the extension `popup` and you will see an installation prompt.
  - Or, just save scripts directly to the Userscripts directory.
- I would suggest changing the reference to “the containing app” to just “the app” since there is only one app, it doesn't need a qualifier.

Thanks! Userscripts is a great piece of software, and I'm super happy to have discovered it! 🙏

**Reply by ACTCD** [reply]
Yes, it's always good to update the docs, I'll do that. Thanks for the feedback and summary.

**Reply by ACTCD** [reply]
addressed in https://github.com/quoid/userscripts/pull/656
